client/src/components/Sidebar/UserList/index.tsx

```diff
@@ -1,11 +1,25 @@
-import React, { useContext } from 'react'
-import { UserInfoContext } from "@/contexts";
+import React, { useContext, useState } from 'react'
+import { UserInfoContext, SocketContext } from "@/contexts";
 import { UserInfo } from "@/contexts/UserInfoContext";
 import AvatarDOM from "@/components/AvatarCanvas/AvatarDOM"
+import { SIOChannel } from "@/contexts/SocketIOContext";
 
 const UserList = () => {
   // TODO: Need to have selfUserInfo here as well
-  const { userInfos } = useContext(UserInfoContext);
+  const { userInfos, editUserName } = useContext(UserInfoContext);
+  const { socket } = useContext(SocketContext);
+
+  socket.on(SIOChannel.EDIT_USER_NAME, ({ id, name }) => {
```

Comment:
Since this event listener will accumulate, it'll be triggered multiple times.
I think you need to put the `socket.on(...)` within the RoomPage `useEffect` hook like:
```
useEffect(() => {
    socket.on(...);
}, []);
```
to only create one listener when the page first renders (attaching it to the UserList would cause it to be created every time the sidebar is re-opened).

Reply:
Ah, got it. Thanks!